errors/no-document-viewport-meta.md

```diff
@@ -0,0 +1,45 @@
+# Viewport `meta` tags should not be used in `_document.js`'s `<Head>`
+
+#### Why This Error Occurred
+
+Adding `<meta name="viewport" ...>` in `pages/_document.js` will lead to unexpected results since the viewport is handled by `next/head`.
+
+#### Possible Ways to Fix It
+
+Set your viewport `meta` tag in `pages/_app.js` instead:
+
+```js
+// pages/_app.js
+import App from 'next/app'
+import Head from 'next/head'
+import React from 'react'
+
+export default class MyApp extends App {
+  static async getInitialProps({ Component, ctx }) {
+    let pageProps = {}
+
+    if (Component.getInitialProps) {
+      pageProps = await Component.getInitialProps(ctx)
+    }
+
+    return { pageProps }
+  }
+
+  render() {
+    const { Component, pageProps } = this.props
+
+    return (
+      <>
+        <Head>
+          <meta name="viewport" content="viewport-fit=cover" />
+        </Head>
+        <Component {...pageProps} />
+      </>
+    )
+  }
+}
```

Comment:
This needs to be updated to the modern custom app design:
https://nextjs.org/docs/advanced-features/custom-app